src/libpmemset/pmemset.c

```diff
@@ -7,23 +7,100 @@
 
 #include "pmemset.h"
 #include "libpmemset.h"
+#include "part.h"
+#include "pmemset_utils.h"
+#include "ravl_interval.h"
+#include "ravl.h"
+#include "alloc.h"
 
 /*
- * pmemset_new -- not supported
+ * mapping_min
+ */
+static size_t
+mapping_min(void *addr)
+{
+	struct pmemset_part_map *pmap = (struct pmemset_part_map *)addr;
+	return (size_t)pmap->addr;
+}
+
+/*
+ * mapping_max
+ */
+static size_t
+mapping_max(void *addr)
+{
+	struct pmemset_part_map *pmap = (struct pmemset_part_map *)addr;
+	return (size_t)pmap->addr + pmap->length;
+}
+
+/*
+ * pmemset_new_init -- initialize set structure.
+ */
+int
+pmemset_new_init(struct pmemset *set, struct pmemset_config *config)
+{
+	ASSERTne(config, NULL);
+
+	/* copy config */
+	memcpy(&set->config, config, sizeof(*config));
+
+	/* intialize RAVL */
+	set->part_map_tree = ravl_interval_new(mapping_min, mapping_max);
+
+	if (set->part_map_tree == NULL) {
+		ERR("ravl tree initialization failed");
```

Comment:
ERR is set by ravl_interval_new, do not override it

Reply:
ERR is NOT set by ravl_interval_new, so I added it in pmemset